Src/Particle/AMReX_ParticleContainerI.H

```diff
@@ -701,87 +722,142 @@ ParticleContainer<NStructReal, NStructInt, NArrayReal, NArrayInt, Allocator>
             virts_offset += np;
         }
     }
-
     if (aggregation_type == "Cell")
     {
+        //Components would be based on
+        MultiFab mf;
+        int nComp = AMREX_SPACEDIM + NStructReal + NArrayReal;
+        // NArrayReal, NStructInt, NArrayInt behavior as before
+        int nGhost = 0;
+        mf.define(ParticleBoxArray(level), ParticleDistributionMap(level), nComp, nGhost);
+
+        iMultiFab imf;
+        nComp = 1 + NStructInt + NArrayInt;
+        imf.define(ParticleBoxArray(level), ParticleDistributionMap(level), nComp, nGhost);
+
+        const auto& geom = Geom(level);
+        const auto plo = geom.ProbLoArray();
+        const auto dxi = geom.InvCellSizeArray();
+        const auto domain = geom.Domain();
+
         BoxList bl_buffer;
         bl_buffer.complementIn(Geom(level).Domain(), ParticleBoxArray(level));
         BoxArray buffer(std::move(bl_buffer));
         buffer.grow(aggregation_buffer);
+        Gpu::synchronize();
```

Comment:
Are you certain that you need all these synchronize() calls? I think all three can be removed, but you should test to make sure.

Reply:
These synchronize calls were for sanity, now that I'm using DeviceVector instead of Vector I can't reproduce the failure I thought these might guard against.